Lib/test/test_dict.py

```diff
@@ -290,6 +290,38 @@ def badgen():
             ['Cannot convert dictionary update sequence element #0 to a sequence'],
         )
 
+    def test_update_shared_keys(self):
+        class MyClass: pass
+
+        # Subclass str to enable us to create an object during the
+        # dict.update() call.
+        class MyStr(str):
+            def __hash__(self):
+                return super().__hash__()
+
+            def __eq__(self, other):
+                # Create an object that shares the same PyDictKeysObject as
+                # the dict
```

Comment:
I'm confused by this comment and the blurb. I think the instances of `MyClass` (`obj` and `obj2`) share the same keys object, but that keys object is not shared with `x`.

Reply:
I updated the comment to make it explicit that it's the same keys object as `obj.__dict__`.

The mutation check is on the argument to `dict.update()`. I don't think there's any check on the receiver.